.github/workflows/test-and-publish.yaml

```
@@ -35,17 +40,28 @@ jobs:
           key: ${{ runner.os }}-spidermonkey
       - name: Build-Library
         if: ${{ steps.cache-spidermonkey.outputs.cache-hit != 'true' }}
-        run: ./setup.sh && ./build_script.sh
+        run: ./setup.sh #&& ./build_script.sh
+      - name: Build wheel
+        run: |
+          echo $(poetry run python --version)
+          poetry build --format=wheel
+          ls -lah ./dist/
       - name: google-tests
         run: |
           cd build
           make && make tests
           gcovr --xml-pretty --exclude-unreachable-branches --print-summary -o coverage.xml
           cd tests && ctest
-          cd ../src
-          python -m pytest ../../tests/python
+          cd ../../
+          poetry run python -m pip install ./dist/*.whl
+          poetry run python -m pytest tests/python
       - name: google-tests-artifacts
         uses: actions/upload-artifact@v3
         with:
           name: ${{ github.job }}-${{ github.run_id }}-${{ github.sha }}
           path: ./build/coverage.xml
+      - name: Buid and Publish package
+        if: ${{ success()  && github.event_name == 'push' && contains(github.ref, 'refs/tags/') }}
+        run: | 
+          echo $(poetry run python --version)
+          poetry publish --no-interaction --skip-existing --build --username __token__ --password $PYPI_API_TOKEN  
```

```suggestion
          poetry publish --no-interaction --skip-existing --build --username __token__ --password {{ secrets.PYPI_API_TOKEN }}
```
action secrets can't be used directly as env vars

---

Thank you! Good catch!